source/particle/property/interface.cc

```diff
@@ -643,6 +643,44 @@ namespace aspect
 
 
 
+      template <int dim>
+      template <typename ParticlePropertyType>
```

Comment:
I think these functions need to be in the header file for the compiler to know which types to instantiate for.

Reply:
It seemed to work without it, but I have no objections to put it in the header file :)

Reply:
I had an error in the plugins I was using this in which made that I actually didn't use the function, so it may not have worked. Anyway, it does seem to work now.